package/MDAnalysis/lib/c_distances.pyx

```diff
@@ -267,3 +271,113 @@ def contact_matrix_pbc(coord, sparse_contacts, box, cutoff):
+# The function in calc_distances.h however assumes that vectors are at most a
+# single box length, so this is modified to first fulfill that assumption
+@cython.boundscheck(False)
+@cython.wraparound(False)
+cdef inline void _minimum_image_triclinic(cython.floating[:] dx, cython.floating[:] box, cython.floating[:] inverse_box):
+    cdef cython.floating dx_min[3]
+    cdef cython.floating s, dsq, dsq_min, rx
+    cdef cython.floating ry[2]
+    cdef cython.floating rz[3]
+    cdef int ix, iy, iz
+
+    # first make shift only 1 cell in any direction
+    s = cround(inverse_box[2] * dx[2])
+    dx[0] -= s * box[6]
+    dx[1] -= s * box[7]
+    dx[2] -= s * box[8]
+    s = cround(inverse_box[1] * dx[1])
+    dx[0] -= s * box[3]
+    dx[1] -= s * box[4]
+    s = cround(inverse_box[0] * dx[0])
+    dx[0] -= s * box[0]
+
+    if cython.floating is float:
+        dsq_min = FLT_MAX
+    else:
+        dsq_min = DBL_MAX
+    dx_min[0] = 0.0
+    dx_min[1] = 0.0
+    dx_min[2] = 0.0
+
+    # then check all images to see which combination of 1 cell shifts gives the best shift
+    for ix in range(-1, 2):
+        rx = dx[0] + box[0] * ix
+        for iy in range(-1, 2):
+            ry[0] = rx + box[3] * iy
+            ry[1] = dx[1] + box[4] * iy
+            for iz in range(-1, 2):
+                rz[0] = ry[0] + box[6] * iz
+                rz[1] = ry[1] + box[7] * iz
+                rz[2] = dx[2] + box[8] * iz
+                dsq = rz[0] * rz[0] + rz[1] * rz[1] + rz[2] * rz[2]
+                if (dsq < dsq_min):
+                    dsq_min = dsq
+                    dx_min[0] = rz[0]
+                    dx_min[1] = rz[1]
+                    dx_min[2] = rz[2]
+
+    dx[0] = dx_min[0]
+    dx[1] = dx_min[1]
+    dx[2] = dx_min[2]
+
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+@cython.cdivision(True)
+def _minimize_vectors_ortho(cython.floating[:, :] vectors not None, cython.floating[:] box not None,
+                            cython.floating[:, :] output not None):
+    cdef int i, n
+    cdef cython.floating box_inverse[3]
+    cdef cython.floating[:] box_inverse_view
+
+    box_inverse[0] = 1.0 / box[0]
+    box_inverse[1] = 1.0 / box[1]
+    box_inverse[2] = 1.0 / box[2]
+
+    box_inverse_view = box_inverse
+    
+    n = len(vectors)
+    for i in range(n):
+        output[i, 0] = vectors[i, 0]
+        output[i, 1] = vectors[i, 1]
+        output[i, 2] = vectors[i, 2]
+        _minimum_image_orthogonal(output[i, :], box, box_inverse_view)
+
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+def _minimize_vectors_triclinic(cython.floating[:, :] vectors not None, cython.floating[:] box not None,
+                                cython.floating[:, :] output not None):
+    cdef int i, n
+    cdef cython.floating box_inverse[3]
+    cdef cython.floating[:] box_inverse_view
+
+    box_inverse[0] = 1.0 / box[0]
+    box_inverse[1] = 1.0 / box[4]
+    box_inverse[2] = 1.0 / box[8]
+
+    box_inverse_view = box_inverse
```

Comment:
Code in calc_distances.h requires 

```
Assumes box having zero values for box[1], box[2] and box[5]:
```
Do we guarantee that through calling convention or do we need to add an explicit check.

Reply:
Taken care of in `check_box`